st2client/st2client/commands/keyvalue.py

```diff
@@ -99,6 +99,8 @@ def __init__(self, resource, *args, **kwargs):
                                  metavar='name',
                                  help='Name of the key value pair.')
         self.parser.add_argument('value', help='Value paired with the key.')
+        self.parser.add_argument('--ttl', type=int, default=None,
+                                 help='TTL for this value.')
```

Comment:
Something to think about is the consistency with st2 token create --ttl. I think that supports `-l`. Please double check and make them consistent.